CHANGES.md

```diff
@@ -1,6 +1,361 @@
+------------------------------------------
+* made `--compiler-driver` the default and added `--no-compiler-driver`  
+  (see https://chapel-lang.org/docs/2.0/technotes/driver.html)
+* added a new flag, `--llvm-print-passes`
+  (see https://chapel-lang.org/docs/2.0/technotes/llvm.html#inspecting-individual-llvm-passes)
+* added more options for the `--llvm-print-ir-stage` flag
+
+Developer-oriented changes: Compiler improvements / changes
+-----------------------------------------------------------
+* improved support for `with` clauses on `foreach` loops
+* fixed a number of bugs with the compiler driver mode
+
+Developer-oriented changes: 'dyno' Compiler improvements / changes
+------------------------------------------------------------------
+* made numerous improvements to the 'dyno' resolver for types and calls:
+  - eliminated incorrect error messages involving type queries
+  - improved default initialization of generics
+  - improved support for `borrowed` and `unmanaged` classes
+  - improved support for interaction between variable arguments and tuples
+* added support for param-folding `select`s w.r.t. copy elision & split-init
+* added support for resolving calls to the `this` method of a field
+* improved the prototype support for library files:
+  - added the ability to reuse LLVM IR stored in a library file
+  - changed to using an ID-based munging strategy for library files
+  - resolve all concrete functions when generating a library file
+* made calling `.type` on a type emit an error
+* implemented type resolution of module-level split-initialized variables
+* implemented `is [const] copyable` and `is [const] assignable` primitives
+* implemented `pragma "last resort"` function resolution logic
+* fixed resolution of `eltType` queries on the `c_ptr` type
+* fixed a bug in resolving user-defined initializers of owned classes
+
+Developer-oriented changes: GPU support
+---------------------------------------
+* added an initial capability to use distributed arrays within GPU kernels
+* improved kernel launch w.r.t. debuggability and to enable upcoming features
+
+
+Developer-oriented changes: Runtime improvements
+------------------------------------------------
+
+Developer-oriented changes: Platform-specific bug fixes
+-------------------------------------------------------
+* removed a developer assertion when an invalid `pragma` was used
+
+Developer-oriented changes: Testing System
+------------------------------------------
+* added a CI check for undocumented stable standard library symbols
+* updated the Python package versions used by `start_test`
+
+Developer-oriented changes: Tool Improvements
+---------------------------------------------
+* added the remaining AST accessors to Dyno's Python bindings
+* improved the Dyno Python bindings to be portable to older Python versions
+* removed the need for the Dyno Python bindings module `chapel.core`
+* added a `clean` target for Dyno's Python bindings
```

Comment:
I wonder if these should be listed under `Developer-oriented changes: 'dyno' Compiler improvements / changes` ? Anyway, it's more accurately called "the compiler frontend library's Python bindings"

This section could be written as

```
* significantly improved the compiler frontend library's Python bindings
  - added the remaining AST accessors
  - improved the portability to older Python versions
  - removed the need for the `chapel.core` module
  - added a `clean` build target
```